k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/selectedstate/command/SelectedStateCommand.java

```diff
@@ -0,0 +1,296 @@
+package com.fsck.k9.mail.store.imap.selectedstate.command;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.store.imap.ImapConnection;
+import com.fsck.k9.mail.store.imap.ImapFolder;
+import com.fsck.k9.mail.store.imap.ImapResponse;
+import com.fsck.k9.mail.store.imap.ImapUtility;
+import com.fsck.k9.mail.store.imap.selectedstate.response.SelectedStateResponse;
+
+
+//This is the base class for a command that is used in the "selected state" i.e when a mailbox is selected
+abstract class SelectedStateCommand {
+
+    /*These limits are 20 octets less than the recommended limits, in order to compensate for the length of the command
+    tag, the space after the tag and the CRLF at the end of the command (these are not taken into account when
+    calculating the length of the command)
+     */
+    private static final int LENGTH_LIMIT_WITHOUT_CONDSTORE = 980;
+    private static final int LENGTH_LIMIT_WITH_CONDSTORE = 8172;
```

Comment:
I disagree. Also I think it's worth mentioning the point in the RFC where the limit is defined.